x-pack/packages/kbn-elastic-assistant/impl/assistant_context/types.tsx

```diff
@@ -15,7 +15,8 @@ export interface MessagePresentation {
 }
 export interface Message {
   role: ConversationRole;
-  content: string;
+  reader?: ReadableStreamDefaultReader<Uint8Array>;
+  content?: string;
```

Comment:
Is this interface intended to stay the same through part2? Will we support streaming turned on/off (even if just technically and not exposed via the UI)? If so, we might want to rework these types so one of the two are required. Does a message make sense without content now? I suppose the new empty 'loading' message would fit this interface?

Reply:
if a message has both content + reader, we know to show the "Regenerate response" button. @jamesspi thinks we should turn streaming on 100% of the time. Even with streaming, once a reader is complete the message will get its `content` property